### PR TITLE
compaction: implement unchecked_tombstone_compaction

### LINKED ISSUE #1487: Add support to compaction properties that relate to tombstone
Those are:

tombstone_compaction_interval   default:864000 (one day)    The minimum number of seconds after an SSTable is created before Cassandra considers the SSTable for tombstone compaction. Tombstone compaction is triggered if the number of garbage-collectable tombstones in the SSTable is greater than tombstone_threshold.

tombstone_threshold default:0.2 The ratio of garbage-collectable tombstones to all contained columns. If the ratio exceeds this limit, Cassandra starts compaction on that table alone, to purge the tombstones.

unchecked_tombstone_compaction   default:false  True allows Cassandra to run tombstone compaction without pre-checking which tables are eligible for this operation. Even without this pre-check, Cassandra checks an SSTable to make sure it is safe to drop tombstones.

### PR DESCRIPTION
**Pull request by nodep**
This change adds the missing Cassandra compaction option unchecked_tombstone_compaction.
Setting this option to true causes the compaction to ignore tombstone_threshold, and decide whether to do a compaction only based on the value of tombstone_compaction_interval

Fixes #1487

### CODE REVIEW

File: docs/troubleshooting/pointless-compactions.rst [hunk 1]
```diff
@@ -60,4 +60,11 @@ For example this will change it to 4 days:
     ALTER table <your table name> with compaction = { 'class' : 'SizeTieredCompactionStrategy', 'tombstone_compaction_interval' : 345600 };
 
 
+2. Make sure `unchecked_tombstone_compaction` is set to false. If it is set to true, `tombstone_threshold` will be ignored,
```
Comment (tchaikov) [inline docs/troubleshooting/pointless-compactions.rst:63]: probably could just use present tense here. see https://developers.google.com/style/tense .

**Review by raphaelsc — APPROVED** [review]
looks good.

I think in practice this option is escalating tombstone compaction to almost major, and it's good that you still keep interval with this option set to avoid compaction loop over the same data. 

Note we also need to bring this to ICS (enterprise only).

**Review by denesb — APPROVED** [review]

**Comment by bhalevy** [comment]
> This change adds the missing Cassandra compaction option unchecked_tombstone_compaction. Setting this option to true causes the compaction to ignore tombstone_threshold, and decide whether to do a compaction only based on the value of tombstone_compaction_interval

Does this fix scylladb/scylladb#1487 or just references it?
In any case, please add the respective github keyword in the cover letter (the description of this PR)
(see https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/using-keywords-in-issues-and-pull-requests)

**Comment by raphaelsc** [comment]
> > This change adds the missing Cassandra compaction option unchecked_tombstone_compaction. Setting this option to true causes the compaction to ignore tombstone_threshold, and decide whether to do a compaction only based on the value of tombstone_compaction_interval
> 
> Does this fix #1487 or just references it? In any case, please add the respective github keyword in the cover letter (the description of this PR) (see https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/using-keywords-in-issues-and-pull-requests)

this was the last option to be implemented, so we can close it with this pr.

**Comment by nodep** [comment]
> > This change adds the missing Cassandra compaction option unchecked_tombstone_compaction. Setting this option to true causes the compaction to ignore tombstone_threshold, and decide whether to do a compaction only based on the value of tombstone_compaction_interval
> 
> Does this fix #1487 or just references it? In any case, please add the respective github keyword in the cover letter (the description of this PR) (see https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/using-keywords-in-issues-and-pull-requests)

I've edited the cover letter.

**Pull request merged**